mantidproject/mantid
# Allow workbench to load the workspaces from mantidplot projects

**Description of work.**

Allows workbench to load the workspaces from mplot projects.
If the file fails to load as JSON, then it will fall back to a second option of selectively parsing out the workspace liost from a mantidplot project.

It does not attempt to do graphs or anything else from the project.  Also qitplot tables will not be recovered.

**Report to:** @AnthonyLim23 


**To test:**
1. Create a project in mantidplot and save it, not the workspaces it includes.
1. Try to load it in workbench, the workspaces should load.

Fixes #27766 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
